frontend/device/input.lua

```diff
@@ -341,6 +341,14 @@ function Input:handleKeyBoardEv(ev)
         return
     end
 
+    -- A Button to turn on/off the frontlight
+    if keycode == "FLSwitch" and ev.value == EVENT_VALUE_KEY_RELEASE then
```

Comment:
I'm afraid I don't understand. This detects a keycode and calls `powerd:detectedFrontlightSwitchToggle()` which indeed generates a key event that is detected by this¿?

Reply:
Yes. 
1) Assume KOreader is started with HardwareSwitch set to on. In KOReader we toggle FL off. Then someone toggles the Hardware Switch on the Device -> FL is set to off by hardware and not toggled on again.
Therefore if a Hardware keypress (FLSwitch) is detected by KOReader AND we know the HardwareSwitch is off then we send another keypress and handle is_fl_on to on.
2) Assume KOReader all intensity on. Turn off the HardwareSwitch. We check that in the first `    if self.is_fl_sw_on and not android:getFrontlightSwitchState() then` and send another key to the system (Hardware light is on, but we want no intensity -> therefore we can set is_fl_on to off
1+2) can be combined with `is_fl_on = not is_fl_on`
At the end of 1+2 we are sure SoftwareSwitch is on, and is_fl_on shows KOReaders toggle state

3) Assume KOReader is started with HardwareSwitch set to off. If we want to change the intensity, we send a keypress to the system (this turns Hardware-Lighting on) and unfortunately we get it also in KOReader as a keyevent. In this case we just have to toggle is_fl_sw_on 

As we might have touched `is_fl_on` we have to set the correct android intensity at the end of the method. And broadcast a changedLightLevel.